Personas (#37)
joomla/rfc: proposed/personas-meta.md

# Personas Meta Document

## 1. Summary

> Personas are fundamentally intended to provide meaningful information for design decisions.

Reviewer: ```suggestion
Personas are fundamentally intended to provide meaningful information for design and development decisions.
```

However, they can also have a broader use in an organisation to understand users and their challenges. Some of the
original motivations for creating personas were to bring empathy for users into the design process, especially in
organisations that are more technically driven. Product management, documentation, marketing, onboarding - these are all
areas where personas can be useful. The key is to construct them in a way that is clear, simple and compelling.

## 2. Why Bother?

Scratching our own itches, it happens to all of us again and again that we overlook where others have their pain points.
We've been discussing how to solve this for a long time and have come to the conclusion that we finally need personas
that represent our users.

## 3. Scope

### 3.1 Goals

The aim is to provide all stakeholders with a means of assessing the consequences of changes to the Joomla CMS on the
various user groups.

### 3.2 Non-Goals

It is not the goal to determine **the** target group for the Joomla CMS. The proportion of users who correspond to a
particular persona must be determined in a subsequent step and then again and again at regular intervals (say annually).

## 4. Approaches

> Joomla is already using kind of personas on the website. The problem is that they are not really personas, but rather:
> <!-- Document: Copy of User Profiles - from Marketing team -->

Reviewer: ```suggestion
Joomla is already using kind of personas on the website. The problem is that they are not really personas, but introductory guides to using Joomla based on how an individual self-identifies among these groups:
<!-- Document: Copy of User Profiles - from Marketing team -->
```

* [Beginner](https://docs.joomla.org/Portal:Beginners)
* [Developer](https://docs.joomla.org/Portal:Developers)
* [Web Developer](https://docs.joomla.org/Web_developers)
* [Site Administrator](https://docs.joomla.org/Portal:Administrators)
* [Web Designer](https://docs.joomla.org/Web_designers)
* [Evaluator](https://docs.joomla.org/Evaluators)
* [Content Creator](https://docs.joomla.org/Content_creators)
* [Publisher](https://docs.joomla.org/Publishers)

### 4.1 Behavioral-based Personas

In the classical understanding, personas illustrate typical representatives of a target group. They have expectations,
values, wishes and goals and show human behaviour. A persona is the personification or prototype of a target group and
helps to make assumptions about customers.

As fictitious users, personas are described with concrete characteristics such as name, photo, place of residence,
curriculum vitae, marital status, age, income, education, skills, attitudes, hobbies, etc. However, they do not
represent the average of a target group, but appear as persons who reveal patterns in their usage behaviour. For this
reason, companies often develop several representatives of a target group.

> Sensis created a set of behavioral-based personas: <!-- Document: Copy of Personas delivered by Sensis -->

Reviewer: Would recommend linking to a different page with these instead of including them here; it's a lot of text that could be overwhelming and distracting from the main point of the document.

* **Web Developer**<br>
  Mike Fenton, 35, is an entrepreneur in Portland, Oregon, who lives with his wife and two kids. He recently left his
  coding job to start his own web development business. An experienced developer, Mike has worked in Drupal and
  WordPress, but prefers Joomla for its extendibility, flexibility, and reliability. As a former systems programmer,
  he’s a supporter of Open Source development platforms, and advocates for Joomla among his peers. Mike resents an
  attitude at Joomla that developers are “a dime a dozen”, but he’s proud that he’s been able to build a business and
  support his family because of Joomla.

* **Web Developer 2**<br>
  Arjuna Patel, 30 years old, is Director of Web Development for an international hotel chain. The son of hotel owners,
  he knows the business, and rapidly rose through the ranks. Arjuna is a Joomla expert and volunteer, active on the
  support forums. He’s currently managing the hotel chain’s transition from Drupal to Joomla, and is frustrated that his
  team consistently has to build extensions to meet business needs, as AOE has done, and hopes Joomla will soon bring
  emphasis to commercial and enterprise solutions.

> Another set of personas was created by Marketing(?): <!-- Document: Copy of Personas for Joomla -->

Reviewer: Same as above; good to have for reference, but not to include all of it in here as these were probably created with a different purpose.

    * MESSAGING
      Modern, expandable, flexible, excellent technical basis, safe, inexpensive implementation, easy to use for the
      customer,
      Open-source - free of charge
    * SOCIAL MEDIA
      XING, LinkedIn, Twitter, FB as an entrepreneur, FB in a private environment

### 4.2 Role-based Personas

In the role-based approach, the focus is on the role of the user. The user is not described as a person, but as a role,
because personal background information such as CV, place of residence, marital status, age or income does not really
help applying the personas. The basis here is the job description, which describes requirements for personality, skills
and knowledge regardless of the person filling the position.

### 4.3 User-Group-based Personas

This is similar to the role-based approach, but the focus is on the user group. User groups are differentiated according
to the way they use Joomla.
For this purpose, the activities in dealing with the CMS are divided into task areas that are as far as possible free of
overlap.
As a result, a real person usually holds more than one of these user roles. On the other hand, it is easier for those

> responsible in the project to assess the impact on the groups.

Reviewer: These are essentially the same. :) Roles are not necessarily jobs or positions, though in many enterprise situations it DOES correlate with job positions.

I'd recommend referring to this approach as role-based to avoid confusion with permission user groups, but we could use user groups as a metaphor to help people understand that while the roles don't usually overlap, individuals can have more than one role.

### 4.4 Chosen Approach

> The classical role-based approach contains much more information than needed for the purpose of this project.
> Moreover, the detailed description of persons risks containing discriminatory elements, starting with the choice of
> gender of a persona.
> In the role-based approach, the tasks overlap, as the division is made according to corporate aspects and not according
> to the concerns of the CMS.
> The user-group-based approach is more suitable for the purpose of this project, as it makes it easy for the stakeholders
> to assess the impact on the groups.

Reviewer: ```suggestion
The classical profile-based approach contains much more information than needed for the purpose of this project.
Moreover, the detailed description of persons risks containing discriminatory elements, starting with the choice of
gender of a persona.
In the role-based approach is more suitable for the purpose of this project, as it makes it easy for the stakeholders
to assess the impact on individuals in each role.
```

### 4.5 Next Steps

Once these personas are ready, the meta-document template for the RFC process should be extended by a
section in which the consequences (pro/con) for the individual groups can be gathered and documented.

> ## 5. Design Decisions
> 
> In the design, it can be assumed that everyone wants to complete their respective task with minimal effort. The type of
> usage is practically predetermined by the user role. Motivation and need are not unimportant dimensions, but they are
> not as decisive as knowledge and skills, which must be taken into account accordingly in the personas.

Reviewer: ```suggestion
## 5. Design and Development Decisions

It can be assumed that everyone wants to complete their respective task with minimal effort. The type of
usage is generally predetermined by the user role. Each persona will have different levels of knowledge, skills, as well as motivations which will help us understand the relative importance of each task and how much effort each person will feel is reasonable.
```

Reviewer: Strongly recommend we don't downplay motivations; it makes a big difference. Someone developing an extension for fun will have a different approach or prioritize different things than someone trying desperately to get something to work for their job.
